anvil/src/eth/backend/mem/mod.rs

```diff
@@ -2257,10 +2254,6 @@ pub fn transaction_build(
                 base_fee.checked_add(max_priority_fee_per_gas).unwrap_or_else(U256::max_value),
             );
         }
-    } else if !is_eip1559 {
-        transaction.max_fee_per_gas = None;
-        transaction.max_priority_fee_per_gas = None;
-        transaction.transaction_type = None;
```

Comment:
I think I now know why we did this:

https://github.com/foundry-rs/foundry/blob/e05b9c75b4501d5880764948b61db787f3dd7fe0/anvil/core/src/eth/transaction/ethers_compat.rs#L109-L111

I don't want to mess with this conversion,

so we should keep this else branch, but we no longer require the `if !is_eip1559`

since we previously always called this with true, I think this is technically a bug fix